sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/SecretKey.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+
+namespace Azure.Security.KeyVault.Keys.Cryptography
+{
+    internal class SecretKey : Key
+    {
+        public SecretKey()
+        {
+            KeyMaterial = new JsonWebKey();
+            KeyMaterial.KeyType = KeyType.Oct;
+            KeyMaterial.KeyOps = new[] { KeyOperation.Encrypt, KeyOperation.Decrypt, KeyOperation.WrapKey, KeyOperation.UnwrapKey };
+        }
+
+        private const string IdPropertyName = "id";
+        private const string ValuePropertyName = "value";
+        private const string ContentTypePropertyName = "contentType";
+
+        public string ContentType { get; private set; }
+
+        internal override void ReadProperty(JsonProperty prop)
+        {
+            switch (prop.Name)
+            {
+                case IdPropertyName:
+                    KeyMaterial.KeyId = prop.Value.GetString();
+                    Properties.Id = new Uri(KeyMaterial.KeyId);
+                    KeyVaultIdentifier kvid = KeyVaultIdentifier.Parse(Properties.Id);
+                    Properties.Name = kvid.Name;
```

Comment:
Rather than open these up as `internal`, what about just doing this all in `ParseId`?